cmd/namespace-lock.go

```diff
@@ -152,9 +162,37 @@ func (n *nsLockMap) lock(volume, path string, lockSource, opsID string, readLock
 
 	// Locking here can block.
 	if readLock {
-		nsLk.RLock()
+		locked = nsLk.GetRLock(timeout)
 	} else {
-		nsLk.Lock()
+		locked = nsLk.GetLock(timeout)
+	}
+
+	if !locked { // We failed to get the lock
+		n.lockMapMutex.Lock()
+		defer n.lockMapMutex.Unlock()
+		// Changing the status of the operation from blocked to none
+		if err := n.statusBlockedToNone(param, lockSource, opsID, readLock); err != nil {
+			errorIf(err, "Failed to clear the lock state")
+		}
+
+		nsLk.ref-- // Decrement ref count since we failed to get the lock
+		// delete the lock state entry for given operation ID.
+		err := n.deleteLockInfoEntryForOps(param, opsID)
+		if err != nil {
+			errorIf(err, "Failed to delete lock info entry")
```

Comment:
we could add `params` and `opsID` to the error log. It would come handy during debugging.